### PR TITLE
maxwell3d: Implement Conditional Rendering

### PR DESCRIPTION
**Pull request by FernandoS27**
Conditional Rendering takes care of conditionaly clearing or drawing
depending on a set of queries. This PR implements the query checks to
stablish if things can be rendered or not.

### CODE REVIEW

File: src/video_core/engines/maxwell_3d.h [hunk 1]
```diff
@@ -89,13 +89,25 @@ class Maxwell3D final {
 
         enum class QuerySelect : u32 {
             Zero = 0,
+            TimeElapsed = 2,
```
Comment (Hexagon12) [inline src/video_core/engines/maxwell_3d.h]: Please list these in order. I've also heard that query mode 26 is also known (transform feedback related), add that too.

File: src/video_core/engines/maxwell_3d.cpp [hunk 2]
```diff
@@ -342,6 +347,43 @@ void Maxwell3D::ProcessQueryGet() {
...
+        break;
+    }
+    case Regs::ConditionMode::ResNonZero: {
+        std::array<u32, 2> data;
+        memory_manager.ReadBlockUnsafe(condition_address, &data, sizeof(data));
+        execute_on = data[0] != 0U && data[1] != 0U;
+        break;
+    }
+    case Regs::ConditionMode::Equal: {
+        std::array<u32, 6> data;
```
Comment (ReinUsesLisp) [inline src/video_core/engines/maxwell_3d.cpp]: If we know what these values mean, this should be passed as a structure.

File: src/video_core/engines/maxwell_3d.h (hunk 1 above)
Comment (ReinUsesLisp) [reply src/video_core/engines/maxwell_3d.h]: I'm fine with just ordering it. We don't exactly know what query 26 is and if we can stub it or not.

File: src/video_core/engines/maxwell_3d.cpp (hunk 2 above)
Comment (FernandoS27) [reply src/video_core/engines/maxwell_3d.cpp]: we kinda don't know them, nouveau uses that but I can have a better look.

**Comment by FernandoS27** [comment]
Ok I've addressed the comments. Changes: comparison now uses a struct instead of an array of data from what I understood they were in nouveau, I ordered the query select types and added query 26 which is related to transform feedback.

**Pull request merged**